docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md

```diff
@@ -0,0 +1,101 @@
+---
+title: '2021-02: Property Conversions'
+date: 2021-02-09
+---
+
+# Property Conversions 
+
+## Context
+
+To facilitate the use of dedicated storage variants for persistence of our custom resources, we need to codegen conversion routines that will copy all properties defined on one version of a resource to another version.
+
+Given the way resources evolve from version to version, these conversions need to handle a wide range of conversions between types that are similar, but not identical.
+
+A **primitive type can become optional** in a later version when a suitable default is provided, or when an alternative becomes available.
```

Comment:
Is this an exhaustive list of conversions? I feel like there's a bunch of cases regarding non-primitive types that are missed here? Maybe highlight that this is just the list of primitive type conversions and that the complex conversions are built out of these based on some rules, or something?

Reply:
Not exhaustive; illustrative. Have reworded as suggested.